proposed/vector-instruction-set-default.md

```diff
@@ -0,0 +1,100 @@
+# Target AVX2 in R2R images
+
+[Vector instructions](https://devblogs.microsoft.com/dotnet/performance-improvements-in-net-5/#intrinsics) have become one of the key pillars of performance in the .NET libraries. Because of that, .NET apps light up a greater percentage of the transistors in a CPU/SOC than they previously did. That's progress. However, we've left performance opportunities on the table that we should consider picking up to make .NET apps run faster and more cheaply (in hosted environments).
+
+Related: [Initial JIT support for SIMD](https://devblogs.microsoft.com/dotnet/the-jit-finally-proposed-jit-and-simd-are-getting-married/)
+
+The minimum supported [SIMD instruction set](https://en.wikipedia.org/wiki/SIMD) for and required by .NET is [SSE2](https://en.wikipedia.org/wiki/SSE2). SSE2 predates the x64 processor by a couple years. SSE2 is old! However, since it is the minimum supported SIMD instruction set, the native code present and executed in ready-to-run images is (you guessed it) SSE2-based. That means your machine uses slower SSE2 instructions even if it (and it almost certainly does) supports the newer [AVX2](https://en.wikipedia.org/wiki/Advanced_Vector_Extensions#Advanced_Vector_Extensions_2) instructions.We rely on the JIT to take advantage of newer (and wider) SIMD instructions than SSE2, such as AVX2. The way the JIT does this is good but not aggressive enough to matter for startup or for apps with shorter life spans (measured in minutes).
+
+SSE4 was introduced with the [Intel Core 2](https://en.wikipedia.org/wiki/Intel_Core_(microarchitecture)) launch in 2006, along with x64. That means that all x64 chips are SSE4 capable, and that SSE2 is really just a holdover from our 32-bit computer heritage. That would make SSE2 a poor baseline for x64 software, but a good one for 32-bit.
```

Comment:
I think this is potentially a bit misleading. `AMD64` was the first x86-64 implementation and was released in ~2003 alongside the Opteron lineup. It had SSE and SSE2 as part of its baseline.

`Intel 64` was first released in ~2004 alongside their Xeon (server) and Prescott (desktop) lineups and was referred to as `EM64T` at the time. I think `Core 2` was just the first of the lineup to use the `Intel 64` name.

`SSE4.1` was announced in 2006 but released alongside Penryn in late 2007, which was a 45nm die-shrink off the Core 2. AMD released `SSE4a` at a similar time on their Barcelona micro-architecture.

I don't think it makes too much difference, just want to ensure that people don't accidentally think that `x64` means `SSE3`, SSSE3`, and `SSE4.1` are definitely available.

Reply:
Good call. I'll clarify that in a later update.